frameworks/Java/spring-webflux/src/main/java/benchmark/repository/PgClientDbRepository.java

```diff
@@ -0,0 +1,87 @@
+package benchmark.repository;
+
+import benchmark.model.Fortune;
+import benchmark.model.World;
+import io.reactiverse.pgclient.PgIterator;
+import io.reactiverse.pgclient.PgPool;
+import io.reactiverse.pgclient.Row;
+import io.reactiverse.pgclient.Tuple;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Component
+@Profile("pgclient")
+public class PgClientDbRepository implements DbRepository {
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    private final PgPool pgPool;
+
+    public PgClientDbRepository(PgPool pgPool) {
+        this.pgPool = pgPool;
+    }
+
+    @Override
+    public Mono<World> getWorld(int id) {
+        log.debug("getWorld({})", id);
+        String sql = "SELECT * FROM world WHERE id = $1";
+
+        return Mono.create(sink ->
+                pgPool.preparedQuery(sql, Tuple.of(id), ar -> {
+                    if (ar.failed()) {
+                        sink.error(ar.cause());
+                    } else {
+
+                        final Row row = ar.result().iterator().next();
+
+                        World world = new World(row.getInteger(0), row.getInteger(1));
+                        sink.success(world);
+                        log.debug("New world - id: {}, randomnumber: {}", world.id, world.randomNumber);
+                    }
+                }));
+    }
+
+    private Mono<World> updateWorld(World world) {
+        String sql = "UPDATE world SET randomnumber = $1 WHERE id = $2";
+
+        return Mono.create(sink -> {
+            pgPool.preparedQuery(sql, Tuple.of(world.randomNumber, world.id), ar -> {
+                if (ar.failed()) {
+                    sink.error(ar.cause());
+                } else {
+                    sink.success(world);
```

Comment:
@davidkiss I think you had an error here, the sink.success was called outside this lambda, so the mono was created before the actual update was done in the DB, which caused the ./tfb script to fail when running the verify phase.

Reply:
Good catch! Thanks